### DISCUSSION TITLE
Building on Windows -fPIC issue

### DISCUSSION THREAD

**Discussion by supernova-III** [carbon-language/carbon-lang#3094] [Getting started] [answered]
Hello, I'm trying to build Carbon, but the following error is reported:
```
.\bazel.exe run //explorer -- ./explorer/testdata/print/format_only.carbon

INFO: Invocation ID: 480fab09-7943-4194-8a7e-f13533804373
INFO: Analyzed target //explorer:explorer (1 packages loaded, 2 targets configured).
INFO: Found 1 target...
ERROR: C:/users/belas/_bazel_belas/g5rygytj/external/llvm_zstd/BUILD.bazel:22:11: Compiling lib/common/xxhash.c failed: (Exit 1): clang failed: error executing command (from target @llvm_zstd//:zstd) C:\Program Files\LLVM\bin\clang -no-canonical-prefixes -fcolor-diagnostics -Werror -Wall -Wextra -Wthread-safety -Wself-assign -Wimplicit-fallthrough -Wctad-maybe-unsupported -Wnon-virtual-dtor ... (remaining 37 arguments skipped)
clang: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'
ERROR: C:/users/belas/_bazel_belas/g5rygytj/external/llvm_zstd/BUILD.bazel:22:11: Compiling lib/common/debug.c failed: (Exit 1): clang failed: error executing command (from target @llvm_zstd//:zstd) C:\Program Files\LLVM\bin\clang -no-canonical-prefixes -fcolor-diagnostics -Werror -Wall -Wextra -Wthread-safety -Wself-assign -Wimplicit-fallthrough -Wctad-maybe-unsupported -Wnon-virtual-dtor ... (remaining 37 arguments skipped)
clang: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'
Target //explorer:explorer failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.637s, Critical Path: 0.16s
INFO: 15 processes: 15 internal.
FAILED: Build did NOT complete successfully
ERROR: Build failed. Not running target
```

how can I fix this?

**Comment by chandlerc** [comment, marked as answer]
The C++ toolchain configuration we have for Bazel doesn't yet support building on Windows, really sorry.

We'd actually love to support this, but none of the currently active contributors uses Windows enough to have the expertise and experience to get it working. We're actively looking for a contributor interested in Windows support, and comfortable working with Bazel, Clang's native installs on Windows, and MSVC. If you or someone else is interested in contributing to getting all of this working, please let us know and happy to chat on Discord about an approach and such. Hopefully, if we get a contributor up for it, we'll then be able to add Windows to our CI to make sure it stays working, and update our docs to help folks get started.

Until then, you should be able to use WSL to build on Windows using our existing instructions. Several folks have reported pretty good success building with WSL already.

**Comment by kehrazy** [comment]
Windows does not use PIC.

We can use different configurations in .bazelrc, e.g.
```
build:linux --force-pic
```

For now, you can comment out the line above in your `.bazelrc`.
